src/components/appNavigation/stackNavigation.js

```diff
@@ -24,10 +25,15 @@ class AppView extends Component<{ descriptors: any, navigation: any, navigationC
    */
   pop = () => {
     const { navigation } = this.props
-
     const nextRoute = this.stack.pop()
     if (nextRoute) {
-      navigation.navigate(nextRoute)
+      const currentRoute = navigation.state.routes[navigation.state.index].key
+      let { screenStates } = this.state
```

Comment:
This is not copying, it's just referencing `screenStates`.

```suggestion
      const { ...screenStates } = this.state.screenStates
```

Reply:
Changes done